### PR TITLE
Render image automatically if no animation is played in a scene

### LINKED ISSUE #1052: Show helpful error message instead of FileNotFound
## Enhancement proposal
When we create a Manim Scene without any animations, and use `-p` flag, Manim saves the output as an image (since you cannot create a video of 1 frames), but then tries to open the video in system viewer, and throws `FileNotFoundError`. This is a very confusing error, to debug, we could instead show a different error when there are 0 animations and `-p` is used.

eg:
```shell
[MM/DD/YY HH:MM:SS] INFO                                                     scene_file_writer.py:549
                             File ready at path\to\project\media\
                             images\scene\SceneName.png

                    INFO     Rendered SceneName                                          scene.py:175
                             Played 0 animations
Error: You cannot preview video without any animations. Try using --save-as-image or add self.wait() 
to your scene.
```

### PR DESCRIPTION
**Pull request by sparshg**
## Changelog / Overview

- If no animations in scene and asked to preview/render a video, preview/render an image instead of giving error.
- Replace `self.play(Animation())` with `self.add()` in tests (without changing control data)
- Fixed a badly written test (ValueTracker test)

## Explanation for changes
As we already know, there is a lot of confusion when new users try to render scenes without animations and getting weird errors. After the scene is finished, just before writing the file, we can check if there are any animations in the scene, if not it changes `config["save_last_frame"]` to True, and `config["write_to_movie"]` to False.

fixes #1052 #869 and https://github.com/ManimCommunity/manim-website/issues/7

## Tests
Passing

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [x] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [x] The PR title is descriptive enough
- [x] The PR is labeled correctly
- [x] The changelog entry is completed if necessary
- [x] Newly added functions/classes either have a docstring or are private
- [x] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Comment by sparshg** [comment]
Should I remove the condition that the image is only generated if attempting to preview/open-file-location for the video? coz without -p or -f, you will still see that error.

**Comment by huguesdevimeux** [comment]
> Should I remove the condition that the image is only generated if attempting to preview/open-file-location for the video? coz without -p or -f, you will still see that error.

I don't think so, but the error should be suppressed in favor of a logging error message. (I don't think it would fit in this PR though, up to you).

By the way, could you add a log message when an image is displayed by default ?

**Comment by sparshg** [comment]
@huguesdevimeux It already says that preview message
Screenshot from 2021-04-06 02-23-27

**Comment by sparshg** [comment]
Ok I added some logging warnings and also `file_writer.finish()` is not called if no animations were found to suppress the error message

**Comment by sparshg** [comment]
Screenshot from 2021-04-06 12-50-33

**Review by jsonvillanueva — APPROVED** [review]

File: manim/scene/scene.py [hunk 1]
```diff
@@ -187,9 +187,15 @@ def render(self, preview=False):
         # We have to reset these settings in case of multiple renders.
         self.renderer.scene_finished(self)
 
-        logger.info(
-            f"Rendered {str(self)}\nPlayed {self.renderer.num_plays} animations"
-        )
+        # Show info only if animations are rendered or to get image
+        if (
+            self.renderer.num_plays != 0
```
Comment (jsonvillanueva) [inline manim/scene/scene.py:192]:
```suggestion
            self.renderer.num_plays
```

File: manim/renderer/cairo_renderer.py [hunk 2]
```diff
@@ -266,7 +266,17 @@ def update_skipping_status(self):
...
+        if self.num_plays != 0:
+            self.file_writer.finish()
+        elif config["preview"] or config["show_in_file_browser"]:
+            config["save_last_frame"] = True
+            config["write_to_movie"] = False
+            logger.warning("No animations found in the scene. Showing an image instead")
+        elif not (config["save_last_frame"] or config["save_pngs"]):
+            logger.warning(
+                "No animations found in the scene. Use -s flag to produce an image"
+            )
```
Comment (jsonvillanueva) [inline manim/renderer/cairo_renderer.py:279]:
```suggestion
        if self.num_plays:
            self.file_writer.finish()
        else:
            config.save_last_frame = True
            config.write_to_movie = False
            self.update_frame(scene, ignore_skipping=False)
            self.file_writer.save_final_image(self.camera.get_image()
```

**Review by jsonvillanueva — COMMENTED** [review]
On second thought, I don't think the logging should exist. I think the expected behavior for `manim -ql [FILE]` and `manim -pql [FILE]` should be to output _some_ file if possible. There should only be a warning if the user explicitly asks for a filetype that isn't possible (e.g. user passes format option expecting mp4/mov file but there's no animations)

File: manim/renderer/cairo_renderer.py (hunk 2 above)
Comment (sparshg) [reply manim/renderer/cairo_renderer.py:279]: Ok but there is a badly written test which which is failing, ValueTrackerTest in updaters, it doesnt add anything to the scene

Comment (sparshg) [reply manim/renderer/cairo_renderer.py]: so I fixed the test and it will now always render something.

**Comment by behackl** [comment]
The issues with the documentation should now be resolved; it might take a while until it manages to run the build (people are working on too many PRs in parallel, which is nice on another level 😄). If it passes, I believe that this can be merged.

Thank you very much for your contribution; I'll wait with my approval until the pipeline passes.

**Comment by jsonvillanueva** [comment]
Ahh... I had a hunch the fix was something to do with `manim_directive.py`

**Review by jsonvillanueva — APPROVED** [review]

**Review by behackl — APPROVED** [review]

**Pull request merged**